mesonbuild/build.py

```diff
@@ -633,8 +633,11 @@ def construct_id_from_path(subdir: str, name: str, type_suffix: str) -> str:
         return my_id
 
     def get_id(self) -> str:
+        name = self.name
+        if getattr(self, 'name_suffix_set', False):
+            name += '.' + self.suffix
```

Comment:
I think this changes the name of the target in the introspection data. The effect can be seen with `meson compile` command:

```
$ meson compile -C builddir main
INFO: autodetecting backend as ninja

ERROR: Can't invoke target `main`: ambiguous name.Add target type and/or path:
- ./bar/main:executable
- ./bar/main:executable
```

Reply:
Ah you're right. I didn't even know this was a feature. I'll look into it.

Reply:
So the ambiguous name warning is correct and actually expected. If there are multiple targets with the same name (possible before this PR), meson compile will give you that message. However what is *not* correct with this PR is that specifying the exact path doesn't allow this to work like it is supposed to. I'll fix that.